src/components/PaginationRow/PaginationRow.js

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { MenuItem, DropdownButton } from '../../index';
+
+const defaultMessages = {
+  firstPage: 'First Page',
+  previousPage: 'Previous Page',
+  nextPage: 'Next Page',
+  lastPage: 'Last Page',
+  perPage: 'per page',
+  of: 'of'
+};
+
+class PaginationRow extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.perPage = Number(props.perPage);
+    this.totalCount = Number(props.totalCount);
+    this.currentPage = Number(props.currentPage);
+
+    this.state = {
+      pageChangeValue: Number(props.currentPage)
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.currentPage !== nextProps.currentPage) {
+      this.setState({ pageChangeValue: Number(nextProps.currentPage) });
+    }
+  }
+
+  msg(key) {
+    return this.props.messages[key] || defaultMessages[key];
+  }
+
+  totalPages() {
+    return Math.ceil(this.props.totalCount / this.perPage);
+  }
+
+  setPageRelative(diff) {
+    this.setPage(Number(this.props.currentPage) + diff);
+  }
+
+  setPage(page) {
+    if (page !== '' && page > 0 && page <= this.totalPages()) {
+      this.props.onPageSet(Number(page));
+    } else {
+      console.error(`Page '${page}' is out of range.`);
+    }
+  }
+
+  handlePageChange(e) {
+    this.setState({ pageChangeValue: e.target.value });
+  }
+
+  handleFormSubmit(e) {
+    this.setPage(this.state.pageChangeValue);
+    e.preventDefault();
```

Comment:
dont think it matters much (or at all), but I would make sure the browser doesn't submit first :)

Reply:
also, shouldn't we allow to execute a custom action? e.g. actually go fetch new data or something?

Reply:
ya i don't think we need to handle form submit... just handle the callback when the values change...and have the server go fetch it...and the send the props back down.

Reply:
It actually prevents the form from being submitted. I didn't find any way how to use `onChange` only, because it's triggered on every key-up -> you can't enter more than one-digit numbers.

`this.setPage` executes `onPageSet` callback that can be injected via props and used to trigger new data fetch.